### PR TITLE
refactor: [M3-7542] - TagsInput & TagsPanel Storybook v7 Stories

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Small PR to handle `.tsx` v7 stories for `TagsInput` & `TagsPanel`

There is no change that should affect client-side code. This only touches our book.

## Changes  🔄
- New v7 stories for `TagsInput` & `TagsPanel`
- New test for `TagsPanel`
- Move styles in own file (`TagsPanel`)
- Remove stories list within documentation

## Preview 📷
Screenshot 2023-12-06 at 11 42 58

Screenshot 2023-12-06 at 11 42 58

### Verification steps 
- pull code and run `yarn storybook`
- verify http://localhost:6006/?path=/docs/components-tags-tags-input--documentation
- verify http://localhost:6006/?path=/docs/components-tags-tags-panel--documentation
- run `yarn test TagsPanel.test.tsx`

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/.storybook/preview.tsx [hunk 1]
```diff
@@ -89,7 +89,6 @@ const preview: Preview = {
           <Description />
           <Primary />
           <Controls />
-          <Stories />
```
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/.storybook/preview.tsx:92]: I'm actually a bit confused where this shows up 😅  - is this regarding the Documentation section of each story, like here:
image

**Review by coliu-akamai — APPROVED** [review]
✅ tests
✅ storybook works! 

Thanks Alban!

File: packages/manager/.storybook/preview.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/.storybook/preview.tsx:92]: It's to show the list of stories in the Documentation page under the args table

File: packages/manager/src/components/TagsPanel/TagsPanel.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  render: (args) => {
+    const TagsInputWrapper = () => {
+      const [{ tags }, updateArgs] = useArgs();
+      const handleUpdateTags = (selected: string[]) => {
+        return Promise.resolve(updateArgs({ tags: selected }));
+      };
+
+      return (
+        <Box sx={{ height: 300 }}>
+          <TagsPanel {...args} tags={tags} updateTags={handleUpdateTags} />
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/TagsPanel/TagsPanel.stories.tsx:23]: > Currently when it's disabled, we lose the click-X-to-remove functionality for tags, but new tags can still be added (this isn't possible in prod Storybook).

I wasn't able to reproduce this but I remove `align` cause it's used nowhere and not even forwarded

File: packages/manager/src/components/TagsInput/TagsInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import type { TagsInputProps } from './TagsInput';
+import type { Item } from 'src/components/EnhancedSelect/Select';
+
+export const Default: StoryObj<TagsInputProps> = {
+  args: {
+    disabled: false,
+    hideLabel: false,
+    label: '',
+    menuPlacement: 'bottom',
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/TagsInput/TagsInput.stories.tsx:17]: I also wasn't able to reproduce this - menu placement gets toggled as expected. I did improve the styling of the story so the select sits in the middle of the canvas to help viewing it better tho.

File: packages/manager/src/components/TagsPanel/TagsPanel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import React from 'react';
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { TagsPanel } from './TagsPanel';
+
+const queryClient = new QueryClient();
+
+const renderWithQueryClient = (ui: any) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/components/TagsPanel/TagsPanel.test.tsx:11]: Can we type this? Not sure if `JSX.Element` is the best or not

Also, should we move this to `packages/manager/src/utilities/testHelpers.tsx`?

```suggestion
const renderWithQueryClient = (ui: JSX.Element) => {
```

Comment (abailly-akamai) [reply packages/manager/src/components/TagsPanel/TagsPanel.test.tsx]: ah yes i left this one out. sure thing - we can be more specific too using

`React.ReactElement<TagsPanelProps>`

Comment (abailly-akamai) [reply packages/manager/src/components/TagsPanel/TagsPanel.test.tsx]: also moved Tag there - makes sense!

**Pull request merged**